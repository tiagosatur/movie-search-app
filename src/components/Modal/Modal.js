import React from 'react';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faWindowClose } from '@fortawesome/free-solid-svg-icons'
import { colorPallete, rem } from '../../style';

const Modal = (props) => {
    const {
        handleClose, children 
    } = props;
  
    return (
      <ModalWrapper {...props}>
        <ModalMain className="modal-main">
            
            <StyledIcon onClick={handleClose} icon={faWindowClose} />
            
            {children}
        </ModalMain>
      </ModalWrapper>
    );
  };

const StyledIcon = styled(FontAwesomeIcon)`
    color: ${colorPallete.white};
    border: ${rem(1)} solid ${colorPallete.primary};
    border-radius: ${rem(3)};
    cursor: pointer;
    font-size: ${rem(40)};
    position: absolute;
    top: 0;
    right: ${rem(-50)};
`;

const ModalWrapper = styled.div`
    background: rgba(0, 0, 0, 0.1);
    display: ${({ show }) => show ? 'block' : 'none'};
    height: 100%;
    position: fixed;
    top: 0;
    left: 0;
    width:100%;
`;

const ModalMain = styled.div`
    align-items: center;
    display: flex;
    border-radius: ${rem(3)};
    background: ${colorPallete.white};
    height: auto;
    justify-content: center;
    min-height: ${rem(490)};
    padding: ${rem(32)} ${rem(16)};
    top: 50%;
    left: 50%;
    position:fixed;
    transform: translate(-50%,-50%);
    width: 80%;

`;

export default Modal;