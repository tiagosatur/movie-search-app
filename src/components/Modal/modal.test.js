import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Modal from './Modal';

describe('Modal tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Modal handleClose={() =>{}} show={true} > 
          <div>Some modal content</div>
      </Modal>
    ).dive();
  });
  
  it('should render the Modal element', () => {
    expect(wrapper.find('.modal').exists()).toBe(true);
  });

  it('should call Modal snapshot correctly', () => {
    const tree = renderer
        .create(wrapper)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });
})