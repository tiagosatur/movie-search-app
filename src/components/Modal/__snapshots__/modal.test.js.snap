// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal tests should call Modal snapshot correctly 1`] = `
<div
  className="modal sc-bwzfXH ffucN"
>
  <div
    className="sc-htpNat ijNTQU"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-window-close fa-w-16 sc-bdVaJa cNLXWh"
      data-icon="window-close"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div>
      Some modal content
    </div>
  </div>
</div>
`;
